src/browser/api_protocol/transport_strategy/ack.ts

```diff
@@ -15,29 +15,32 @@ limitations under the License.
 */
 
 const errors = require('../../../common/errors');
+import { EverTrue, EverFalse } from '../../../shapes';
 
 export class AckMessage {
     public readonly action: string = 'ack';
     public correlationId: number;
     public payload: AckPayload | NackPayload;
 }
 
+// ToDo following duplicated in src/shapes.ts
+
 export class AckPayload {
-    public readonly success: boolean = true;
-    public data: any;
+    public success: EverTrue;
+    public data?: any;
 
     constructor(data: any) {
         this.data = data;
     }
 }
 
 export class NackPayload {
-    public readonly success: boolean = false;
-    public reason: string = '';
-    public error: Error = null;
+    public success: EverFalse;
+    public reason?: string = '';
```

Comment:
`error` at least should be, per `NackPayload`'s constructor.
The `APIPayloadNack` (in src/shapes.ts) interface suggests `reason` might be too ...
We are considering merging these interfaces, right?